packages/date/src/index.js

```diff
@@ -314,18 +314,27 @@ export function format( dateFormat, dateValue = new Date() ) {
 }
 
 /**
- * Formats a date (like `date()` in PHP), in the site's timezone.
+ * Formats a date (like `date()` in PHP).
  *
- * @param {string}                           dateFormat PHP-style formatting string.
- *                                                      See php.net/date.
- * @param {(Date|string|Moment|null)}        dateValue  Date object or string,
- *                                                      parsable by moment.js.
+ * @param {string}                      dateFormat PHP-style formatting string.
+ *                                                 See php.net/date.
+ * @param {(Date|string|Moment|null)}   dateValue  Date object or string, parsable
+ *                                                 by moment.js.
+ * @param {(string|number|null)}        timezone   Timezone to output result in or
+ *                                                 a UTC offset.
+ *                                                 Defaults to timezone from site.
+ *                                                 See momentjs.com.
  *
- * @return {string} Formatted date.
+ * @return {string} Formatted date in English.
  */
-export function date( dateFormat, dateValue = new Date() ) {
-	const offset = settings.timezone.offset * HOUR_IN_MINUTES;
-	const dateMoment = momentLib( dateValue ).utcOffset( offset, true );
+export function date( dateFormat, dateValue = new Date(), timezone ) {
+	if ( timezone && ! isUtcOffset( timezone ) ) {
+		const dateMoment = momentLib( dateValue ).tz( timezone );
+		return format( dateFormat, dateMoment );
+	}
+
+	const offset = isUtcOffset( timezone ) ? timezone : settings.timezone.offset * HOUR_IN_MINUTES;
```

Comment:
The PR works for me at first glance, _except_ for daylight savings.

I'm guessing this needs to determine the offset on the _given_ date, rather than the _current_ date.

```js
console.log(
	'site current offset: ' + wp.date.__experimentalGetSettings().timezone.offset + "\n",
	'browser current offset: ' + new Date().getTimezoneOffset() / -60 + "\n",
	'past date in standard: ' + wp.date.date( 'Y-m-d h:ia', 1552132800 * 1000 ), "\n",
   	'past date in DST: ' + wp.date.date( 'Y-m-d h:ia', 1566388800 * 1000 ) + "\n"
);
```

Expected:

```
site current offset: -5
 browser current offset: -8
 past date in standard: 2019-03-09 07:00am 
 past date in DST: 2019-08-21 08:00am
```

Actual:

```
site current offset: -5
 browser current offset: -8
 past date in standard: 2019-03-09 07:00am 
 past date in DST: 2019-08-21 07:00am
```

Reply:
You mean when using WordPress’ default timezone, right?

In this case, we need to (a) check if WordPress is using a timezone instead of a UTC offset and, if it does, use it (as timezones are sensitive to daylight savings, but UTC offsets aren't). And, if WordPress is using a UTC offset, then use it _as is_ and forget about daylight savings.

Does this make sense?

Reply:
Do you mean using using `settings.timezone.string` instead of `settings.timezone.offset`? If so, then I think that makes sense 👍 . The site I'm testing with has the `timezone_string` option set to `America/New_York`, rather than `UTC-5`.

Reply:
I'm using this as a temporary workaround in one of my projects, and it seems to be working so far:

```js
export function _tmp_dateI18n( dateFormat, dateValue ) {
	let dateMoment;

	if ( settings.timezone.string ) {
		dateMoment = momentLib( dateValue ).tz( settings.timezone.string );
	} else {
		const offset = settings.timezone.offset * HOUR_IN_MINUTES;
		dateMoment   = momentLib( dateValue ).utcOffset( offset );
	}

	dateMoment.locale( settings.l10n.locale );

	return format( dateFormat, dateMoment );
}
```

I stripped out some stuff that isn't relevant in my usage, but the core logic seems like what you were thinking.

Reply:
Hey @iandunn. Thanks for the feedback and sorry for the delay (I had ankle surgery and was AFK).

I tweaked the source code so that, when no timezone/UTC offset is provided, it first tries to use the site’s timezone (which does take into account daylight savings) and finally defaults to the site’s UTC offset. I think this should fix your issue...

Oh! And I also modified the tests to include your scenario.

Let me know how it works :-)